networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/Eth2NetworkBuilder.java

```diff
@@ -101,7 +101,13 @@ public Eth2NetworkBuilder eventBus(final EventBus eventBus) {
     return this;
   }
 
-  public Eth2NetworkBuilder chainStorageClient(final RecentChainData recentChainData) {
+  public Eth2NetworkBuilder historicalChainData(final StorageQueryChannel historicalChainData) {
+    checkNotNull(historicalChainData);
+    this.historicalChainData = historicalChainData;
+    return this;
+  }
```

Comment:
We talked about using `HistoricalChainData` as a wrapper around `StorageQueryChannel` but I hadn't realised we were able to completely remove `HistoricalChainData`.  So actually I think using the channel directly is better than the wrapper class in this case.

I am however tempted to move as much stuff to use `CombinedChainData` as possible rather than directly requesting data from storage.  Then it can load content from memory when possible and that would be the place we could add caching if needed.  I'd probably only do that in this PR where `CombinedChainData` is already reasonably easily available - otherwise it's just something we'll get to in the future.

Reply:
Discussed offline to move as much stuff to use `CombinedChainDataClient` as possible, but decided to do this after the move to EventChannels is done. Here are the jira tickets for planned work:
- https://pegasys1.atlassian.net/secure/RapidBoard.jspa?rapidView=67&modal=detail&selectedIssue=BC-320
- https://pegasys1.atlassian.net/secure/RapidBoard.jspa?rapidView=67&modal=detail&selectedIssue=BC-321